File: dosomething.js
function ProcessList(list, callback){
  return callback(list);
}

function SumNumbers(list){
  return list.reduce((sum, n) => sum + n, 0);
}

function PrintNumbers(list){
  for (const n of list)
    console.log(n);
}

const data = [3, 4, 1, 9];

// --- Add the values together
sum = ProcessList(data, SumNumbers);
console.log(sum);
console.log("\n");

// --- Print the Numbers
ProcessList(data, PrintNumbers);
console.log("\n");

File: dosomething.cpp
#include <iostream>
#include <list>
#include <functional>

int processList(
    const std::list<int>& data,
    std::function<int(const std::list<int>&)> operation)
{
    return operation(data);
}

int main()
{
    std::list<int> numbers = {1, 2, 3, 4, 5};

    int sum = processList(numbers, [](const std::list<int>& lst) {
        int total = 0;
        for (int x : lst)
            total += x;
        return total;
    });

    processList(numbers, [](const std::list<int>& lst){
	    for (auto x : lst)
	    	std::cout << x << "\t";
	    std::cout << std::endl;
	    return 0;
    });

    std::cout << "Sum: " << sum << std::endl;

    return 0;
}

